llvm/include/llvm/CodeGen/GlobalISel/GenericMachineInstrs.h

```diff
@@ -950,6 +950,16 @@ class GExtOrTruncOp : public GCastOp {
   };
 };
 
+/// Represents a splat vector.
+class GSplatVector : public GenericMachineInstr {
+public:
+  Register getValueReg() const { return getOperand(1).getReg(); }
```

Comment:
To be more descriptive how about call this `getScalarReg()` instead?

Reply:
```
// Generic splatvector.
def G_SPLAT_VECTOR: GenericInstruction {
  let OutOperandList = (outs type0:$dst);
  let InOperandList = (ins type1:$val);
  let hasSideEffects = false;
}
```
In the .td file it is called `val`.  Scalar is confusing for me on a scalable vector.